### PR TITLE
Async support and some rework

### PR DESCRIPTION
**Pull request by imaji**
Updated .NET to 4.6.2
Added initial support for Async Await
Updated dependencies

### CODE REVIEW

File: ExampleApplication/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -6,11 +6,11 @@
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
 [assembly: AssemblyTitle("ExampleApplication")]
-[assembly: AssemblyDescription("")]
+[assembly: AssemblyDescription("An Example App for user with Pusher Client")]
 [assembly: AssemblyConfiguration("")]
-[assembly: AssemblyCompany("Microsoft")]
+[assembly: AssemblyCompany("")]
```
Comment (WillSewell) [inline ExampleApplication/Properties/AssemblyInfo.cs]: `Pusher`?

File: ExampleApplication/packages.config [hunk 2]
```diff
@@ -1,4 +1,23 @@
...
+  <package id="System.Runtime.InteropServices" version="4.3.0" targetFramework="net462" />
+  <package id="System.Security.Cryptography.Algorithms" version="4.3.1" targetFramework="net462" />
+  <package id="System.Security.Cryptography.Encoding" version="4.3.0" targetFramework="net462" />
+  <package id="System.Security.Cryptography.Primitives" version="4.3.0" targetFramework="net462" />
+  <package id="System.Security.Cryptography.X509Certificates" version="4.3.2" targetFramework="net462" />
+  <package id="System.Text.RegularExpressions" version="4.3.0" targetFramework="net462" />
+  <package id="System.Threading" version="4.3.0" targetFramework="net462" />
+  <package id="System.Threading.Timer" version="4.3.0" targetFramework="net462" />
+  <package id="WebSocket4Net" version="0.15.2" targetFramework="net462" />
 </packages>
```
Comment (WillSewell) [inline ExampleApplication/packages.config]: Add newline here

File: PusherClient.Tests.Utilities/Config.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            }
+        }
+
+        public static string AppId { get; set; }
+
+        public static string AppKey { get; set; }
+
+        public static string AppSecret { get; set; }
+    }
+}
```
Comment (WillSewell) [inline PusherClient.Tests.Utilities/Config.cs]: Newline

File: README.md [hunk 4]
```diff
@@ -63,8 +89,19 @@ _pusher.Error += _pusher_Error;
...
+    Cluster = "eu"
+});
+_pusher.ConnectionStateChanged += _pusher_ConnectionStateChanged;
+_pusher.Error += _pusher_Error;
+ConnectionState connectionState = _pusher.ConnectAsync();
+```
+
 ### Subscribe to a public or private channel
 
+####Event based
```
Comment (WillSewell) [inline README.md]: Missing a space

**Comment by imaji** [comment]
Yes, we could do an await next message. But that would put the emphasis on the consumer to await the message after that one, etc.

I'll sort the other newline and space issues out this evening.

**Comment by imaji** [comment]
Didn't mean to close this.

**Comment by WillSewell** [comment]
Fair enough. It looks like we do something similar in the libraries for our new products: we use promises for connecting/subscribing and a callback function for subscription events. At least this is consistent.

**Comment by WillSewell** [comment]
I believe @ewmy still wants to test this PR, but it looks good from my PoV.

**Comment by kn100** [comment]
Tested some basic functionality of this PR, seems to work fine!

**Pull request merged**